consensus_encoding/src/decode/decoders.rs

```diff
@@ -100,13 +64,15 @@ impl<const N: usize> Decoder for ArrayDecoder<N> {
     }
 }
 
-/// A decoder which decodes two objects, one after the other.
-pub struct Decoder2<A, B>
+/// A decoder which wraps two inner decoders and returns the output of both.
+/// The error types of the inner decoders are mapped to a common type.
+pub struct Decoder2<A, B, Err>
 where
     A: Decoder,
     B: Decoder,
 {
     state: Decoder2State<A, B>,
+    _error: core::marker::PhantomData<Err>,
```

Comment:
nit: We can just import the `PhantomData` type, more terse with no loss of clarity. I added #5010 while thinking about this name. FTR I added the `marker` one in `block` which I now see should have had the underscore.